pureconfig/src/main/scala/com/avast/clients/rabbitmq/pureconfig/PureconfigRabbitMQConnection.scala

```diff
@@ -0,0 +1,136 @@
+package com.avast.clients.rabbitmq.pureconfig
+
+import cats.effect.{ConcurrentEffect, Resource}
+import com.avast.clients.rabbitmq.api._
+import com.avast.clients.rabbitmq.{
+  BindExchangeConfig,
+  BindQueueConfig,
+  ChannelListener,
+  ConnectionListener,
+  ConsumerConfig,
+  ConsumerListener,
+  DeclareExchangeConfig,
+  DeclareQueueConfig,
+  DeliveryConverter,
+  DeliveryReadAction,
+  ProducerConfig,
+  ProductConverter,
+  PullConsumerConfig,
+  RabbitMQConnection,
+  ServerChannel
+}
+import com.avast.metrics.scalaapi.Monitor
+import com.typesafe.config.Config
+import pureconfig.ConfigSource
+
+import scala.language.higherKinds
+import scala.reflect.ClassTag
+
+trait PureconfigRabbitMQConnection[F[_]] {
```

Comment:
I don't get this trait. There is nothing PureConfig-related in it except that it uses that library internally. I don't think this should be a trait at all, what you need is an extension method of `RabbitMqConnection` that just loads the configuration using PureConfig and passes the configuration as case class to the underlying "original" implementation. Nothing more. Trait basically represents some kind of abstraction and this is no such thing.

Reply:
Solved in person.